### PR TITLE
Upgrade from our standard lints v4 -> v6

### PR DESCRIPTION
**Pull request by repi**
And switches to enable them in root `.cargo/config.toml` so they are enabled for all crates in the repository. Also resolves some warnings as errors when running clippy on some of the crates that we had.

The code fixups for this was mostly from the [`clippy::single_match_else`](https://rust-lang.github.io/rust-clippy/master/#single_match_else) lint. In our other code bases we've been seeing some good improvements from the style that that enforces as avoids deeper indentation, though in this codebase the both cases look quite similar but probably doesn't hurt to just use our default.

### CODE REVIEW

File: .cargo/config [hunk 1]
```diff
@@ -1,2 +1,85 @@
...
+    "-Wclippy::verbose_file_reads",
+    "-Wclippy::zero_sized_map_values",
+    "-Wfuture_incompatible",
+    "-Wnonstandard_style",
+    "-Wrust_2018_idioms",
+    # END - Embark standard lints v6 for Rust 1.55+
+
+    # repo specific lints
+    "-Aunsafe_code",
+]
```
Comment (eddyb) [inline .cargo/config:85]: Missing newline at the end of file.

**Review by eddyb — CHANGES_REQUESTED** [review]
LGTM, just one tiny nit.

File: .cargo/config (hunk 1 above)
Comment (repi) [reply .cargo/config:85]: hah fixed

**Comment by repi** [comment]
Fixed your nit @eddyb :) 

CI fails, but currently all CI fails on Mac due to #865, so just going to merge this one

**Pull request merged**